src/Compute.js

```diff
@@ -0,0 +1,210 @@
+    } catch (error) {
+      console.error(error.message)
+    }
+  }
+
+  async function startCompute(algorithmId, algorithmMeta) {
+    try {
+      const accounts = await ocean.accounts.list()
+      const ComputeOutput = {
+        publishAlgorithmLog: publishLogState,
+        publishOutput: publishOutputState,
+        brizoAddress: ocean.config.brizoAddress,
+        brizoUri: ocean.config.brizoUri,
+        metadataUri: ocean.config.aquariusUri,
+        nodeUri: ocean.config.nodeUri,
+        owner: accounts[0].getId(),
+        secretStoreUri: ocean.config.secretStoreUri
+      }
+      console.log(ComputeOutput)
+      // order the compute service
+      const agreement = await ocean.compute.order(accounts[0], ddoAssetId)
+      setAgreementId(agreement)
+      // start a compute job
+      const status = await ocean.compute.start(
+        accounts[0],
+        agreement,
+        algorithmId,
+        encodeURIComponent(JSON.stringify(algorithmMeta)),
+        ComputeOutput
+      )
+      setJobId(status.jobId)
+      console.log(status)
+      alert('Job created.  You can query for status now')
+    } catch (error) {
+      console.error(error.message)
+    }
+  }
+
+  // order and start the compute service with an algorithm published as an asset
+  async function startWithPublishedAlgo() {
+    return startCompute(ddoAlgorithmId)
+  }
+
+  // order and start the compute service with a passed raw algorithm
+  async function startWithRawAlgo() {
+    rawAlgoMeta.rawcode = textRawAlgo
+    return startCompute(undefined, rawAlgoMeta)
+  }
+
+  async function getStatus() {
+    try {
+      const accounts = await ocean.accounts.list()
+
+      const status = await ocean.compute.status(accounts[0], agreementId, jobId)
+      setJobStatus(JSON.stringify(status, null, '\t'))
+      console.log(status)
+    } catch (error) {
+      console.error(error.message)
+    }
+  }
+
+  async function showDivAlgo() {
+    setIsAlgoInputVisible(isAlgoInputVisible ? false : true)
+  }
+
+  async function updateRawAlgoCode(event) {
+    setTextRawAlgo(event.target.value)
+  }
+
+  async function updateDdoAssetId(event) {
+    setDdoAssetId(event.target.value)
+  }
+  async function handlePublishOutputState(event) {
+    setPublishOutputState(event.target.checked)
+  }
+  async function handlePublishLogState(event) {
+    setPublishLogState(event.target.checked)
+  }
+
+  if (!web3) {
+    return null
+  }
+
+  return (
+    <>
+      <h3>Compute</h3>
+      <ComputeSection>
+        <button onClick={publish}>Publish dataset with compute service</button>
+        Asset DID:
+        <input type="text" value={ddoAssetId} onChange={updateDdoAssetId} />
+      </ComputeSection>
+      <ComputeSection>
+        <button onClick={publishalgo}>Publish algorithm</button>
+        Algo DID:
+        <input type="text" value={ddoAlgorithmId} readOnly />
+      </ComputeSection>
+      <ComputeSection>
+        <input
```

Comment:
this should be more semantic and accessible with proper `label` for each `input`. There should never be an `input` without a `label`, and label and input need to be connected with `for/name` attributes

Reply:
I'll quickly do the markup changes